infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/CheServerKubernetesClientFactory.java

```diff
@@ -66,7 +66,7 @@ public KubernetesClient create() throws InfrastructureException {
   }
 
   @Override
-  protected Config buildConfig(Config config, String workspaceId) {
+  protected Config buildConfig(Config config, String workspaceId, String token) {
```

Comment:
Can we take the token from EnvironmentContext?

Reply:
probably yes, I'll take a look.

Reply:
reworked, done